js/modules/loki_snode_api.js

```diff
@@ -108,9 +125,12 @@ class LokiSnodeAPI {
 
   async refreshSwarmNodesForPubKey(pubKey) {
     const newNodes = await this.getFreshSwarmNodes(pubKey);
-    await window.Signal.Data.saveSwarmNodesForPubKey(pubKey, newNodes, {
-      Conversation: Whisper.Conversation,
-    });
+    try {
+      const conversation = ConversationController.get(pubKey);
+      await conversation.updateSwarmNodes(newNodes);
+    } catch (e) {
+      throw e;
+    }
```

Comment:
You could simply reuse the `updateSwarmNodes` function you added above:
```js
async refreshSwarmNodesForPubKey(pubKey) {
    const newNodes = await this.getFreshSwarmNodes(pubKey);
    await this.updateSwarmNodes(pubKey, newNodes);
}
```
Also no need to catch then re-throw an error?

Reply:
Yes, you are right!